### PR TITLE
Build system check for cap library and uses it without linking to it

### PR DESCRIPTION
**Pull request by gcsideal**
While packaging libwebsockets 4.1.0 RC1 for Debian, I found that cmake check for cap_set_flag() and uses it if available - but doesn't link to the library. This pull request corrects linking.

### CODE REVIEW

**Comment by lws-team** [comment]
Thanks a lot for the fix, it's pushed on master.  And thanks for the packaging...

**Comment by RSATom** [comment]
I don't sure, but maybe with this patch there are no need to add `libcap` dependency to `libwebsockets.pc` ?
@gcsideal what do you think?

**Comment by RSATom** [comment]
@gcsideal also, what do you think about enable GLIB support in libwebsockets by default in Debian/Ubuntu?

**Comment by lws-team** [comment]
On the distro options, there is a `LWS_WITH_DISTRO_RECOMMENDED` that includes glib and is tested in CI on 8 platforms, eg, f32 riscv, focal and bionic x86_64

https://libwebsockets.org/git/libwebsockets/tree/CMakeLists-implied-options.txt#n76-113

It'd be great if distros could use that since it enables lws use with more complex apps from the package.

**Comment by gcsideal** [comment]
> I don't sure, but maybe with this patch there are no need to add `libcap` dependency to `libwebsockets.pc` ?
> @gcsideal what do you think?

These are different things. Linking is a runtime dependency - that library should be mapped in as well to run the code. libwebsockets.pc needed for providing information via pkg-config for compile (it can give back the needed headers location as well for example) and do the actual linking of an other library or ELF binary.

**Comment by gcsideal** [comment]
> @gcsideal also, what do you think about enable GLIB support in libwebsockets by default in Debian/Ubuntu?
> 
> P.S.: Sorry for offtopic...

Yup, it's off-topic here. Thus in short I have to find the balance between providing a library with several functions but with less dependencies. Question is how many users need GLIB support and how big (in number and in size) dependencies that would pull in for all the users (who probably don't need GLIB support).

**Comment by gcsideal** [comment]
> On the distro options, there is a `LWS_WITH_DISTRO_RECOMMENDED` that includes glib and is tested in CI on 8 platforms, eg, f32 riscv, focal and bionic x86_64
> 
> https://libwebsockets.org/git/libwebsockets/tree/CMakeLists-implied-options.txt#n76-113
> 
> It'd be great if distros could use that since it enables lws use with more complex apps from the package.

Do you have any kind of statistics how many people / other code (would) use GLIB functionality in libwebsockets?

**Comment by lws-team** [comment]
> Do you have any kind of statistics how many people / other code (would) use GLIB functionality in libwebsockets?

Not really... I only get to hear about usage when there are problems.  There's at least one other project using it for an Ardour plugin I worked with them to produce dfed5475295e552bdbf5165193eb7e40ece71a45

https://github.com/warmcat/libwebsockets/issues/1963

I can understand not wanting too many package deps but it's a bit chicken-and-egg.  I guess an answer is bind them as plugins at runtime and package them separately then, but that's not how they are atm.

**Comment by gcsideal** [comment]
> > Do you have any kind of statistics how many people / other code (would) use GLIB functionality in libwebsockets?
> 
> Not really... I only get to hear about usage when there are problems. There's at least one other project using it for an Ardour plugin I worked with them to produce [dfed547](https://github.com/warmcat/libwebsockets/commit/dfed5475295e552bdbf5165193eb7e40ece71a45)
> 
> #1963
> 
> I can understand not wanting too many package deps but it's a bit chicken-and-egg. I guess an answer is bind them as plugins at runtime and package them separately then, but that's not how they are atm.

Yup, a plugin system would be nice. At the moment only one user asked for GLIB support, I guess it's not that commonly used. But prove me wrong, note me projects that use libwebsockets with GLIB bindings.

**Comment by lws-team** [comment]
... nobody's trying to tell you there are many glib / lws users... I know of 2 from interacting with them.  It was only introduced a few months ago just before v4.0 release, and the people interested to use it will typically be gtk projects first, which is not a huge catchment area.  But they do exist same as there are users for all the other event libs too.  And unless they have problems, they won't come here and let me know.

Basically of the distros that carry lws, only gentoo's 'packaging' is flexible enough to reflect the build-time options in lws.  The others are missing out on a lot of stuff.
